rclcpp/include/rclcpp/parameter_events_subscriber.hpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__PARAMETER_EVENTS_SUBSCRIBER_HPP_
+#define RCLCPP__PARAMETER_EVENTS_SUBSCRIBER_HPP_
+
+#include <string>
+#include <utility>
+#include <unordered_map>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/parameter_events_filter.hpp"
+
+namespace rclcpp
+{
+
+class ParameterEventsSubscriber
+{
+public:
+  ParameterEventsSubscriber(
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
+    rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
+    rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
+    const rclcpp::QoS & qos =
+    rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_parameter_events)));
+
+  template<typename NodeT>
+  ParameterEventsSubscriber(
+    NodeT node,
+    const rclcpp::QoS & qos =
+    rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_parameter_events)))
+  : ParameterEventsSubscriber(
+      node->get_node_base_interface(),
+      node->get_node_topics_interface(),
+      node->get_node_logging_interface(),
+      qos)
+  {}
+
+  /// Sets a custom callback for parameter events
+  /**
+   * If no namespace is provided, a subscription will be created for the current namespace
+   * Repeated calls to this function will overwrite the callback
+   * If more than one namespace already has a subscription to its parameter events topic, then the
+   * provided callback will be applied to all of them
+   *
+   * \param[in] callback Function callback to be evaluated on event
+   * \param[in] Name of namespace for which a subscription will be created
+   */
+  void set_event_callback(
+    std::function<void(const rcl_interfaces::msg::ParameterEvent::SharedPtr &)> callback,
+    const std::string & node_namespace = "");
+
+  /// adds a custom callback for a specified parameter
+  /**
+   * If a node_name is not provided, defaults to the current node
+   *
+   * \param[in] parameter_name Name of parameter
+   * \param[in] callback Function callback to be evaluated upon parameter event
+   * \param[in] node_name Name of node which hosts the parameter
+   */
+  void register_parameter_callback(
```

Comment:
I agree that `register` may be more ambiguous. Since in the current state we can only have one callback per param/node and per event, then it makes sense to use `set` for now